### PR TITLE
WIP partial work on using recommended ts config

### PR DESCRIPTION
**Pull request by derduher**
Feel free to comment as I work. 😄

### CODE REVIEW

File: lib/sitemap.d.ts [hunk 1]
```diff
@@ -19,7 +19,7 @@ export declare function createSitemap(conf: {
     cacheTime: number;
     xslUrl: string;
     xmlNs?: string;
-}): Sitemap;
```
Comment (derduher) [inline lib/sitemap.d.ts:22]: this was not an intentional change. I will look.

Comment (derduher) [reply lib/sitemap.d.ts:22]: nothing from this file was intentionally changed. I will revert.

File: lib/sitemap-item.ts [hunk 2]
```diff
@@ -1,22 +1,22 @@
-import ut = require('./utils');
-import fs = require('fs');
-import builder = require('xmlbuilder');
-import isArray = require('lodash/isArray');
+import ut from './utils';
```
Comment (derduher) [inline lib/sitemap-item.ts:1]: Will be using babel to transpile

File: lib/sitemap.ts [hunk 3]
```diff
@@ -27,30 +27,31 @@ import { ICallback, ISitemapImg, SitemapItemOptions } from './types';
...
   urls: (string | SitemapItemOptions)[]
 
   cacheResetPeriod: number;
   cache: string
   xslUrl: string
   xmlNs: string
-  root: builder.XMLElementOrXMLNode & {
-    attribs?: [],
-    children?: [],
+  root: builder.XMLElement & {
```
Comment (derduher) [inline lib/sitemap.ts]: Changed in v13

File: package.json [hunk 4]
```diff
@@ -22,35 +22,87 @@
...
+      "jest": true,
       "node": true
+    },
+    "rules": {
+      "no-case-declarations": 0,
+      "no-console": 0,
+      "no-unused-vars": 0,
+      "react/prop-types": 0,
+      "indent": "off",
+      "no-dupe-class-members": "off",
```
Comment (derduher) [inline package.json:85]: official recommendation for allowing method overloading

**Comment by bluelovers** [comment]
why has `babel.config.js`?

**Pull request merged**